yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -44,23 +44,67 @@ namespace NDetail {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+TSharedRef TruncateRowset(TSharedRange<NTableClient::TUnversionedRow>& rows)
+{
+    if (rows.Empty()) {
+        return TSharedRef::MakeEmpty();
+    }
+
+    // Wire format doesn't provide result size before serializing, so we have to make some estimations and verify them.
+    i64 estimationThresholds[] = {MaxStringValueLength, MaxStringValueLength - 1_MB, 1_MB, (i64)GetDataWeight(rows[0])};
```

Comment:
`(i64)GetDataWeight(rows[0])` won't work as an estimate since for one row it will be `GetUnversionedRowByteSize(rows.Size()) + (i64)GetDataWeight(rows[0])`.